woocommerce/woocommerce-ios
# [Tablet Orders] Reset product selector when modal dismissed

Part of: #11235


## Description


Previously, with the `sideBySideViewForOrderForm` feature flag enabled, changes to product selections in the modally-presented product or variation selector were not reset when the product selector was closed.

This meant that when we next opened the product selector it was out of sync with the order summary, causing confusion. This was exacerbated by the toggle (rather than set) behaviour for changing a product selection.

In this PR, we reset the `productSelectorViewModel` to nil when the modal is dismissed, even when the feature flag is true. The next time we open the product selector, it's created with the up-to-date state from the order.

## Testing instructions


1. Enable the `sideBySideViewForOrderForm` feature flag
2. Launch the app on an iPhone or iPad with a horizontally compact split view
3. Open the Orders tab and tap `+`
4. Tap `Add Products`
5. Select some products in the sheet-presented product selector, but don't confirm
6. Close the product selector
7. Tap `Add Products` again
8. Observe that no products are selected
9. Select and confirm some products
10. Tap `+` to add more products
11. Change the selections, and close without confirming
12. Reopen the product selector; observe that the selections match the order and not your previous changes.

Repeat the above with the feature flag disabled.

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/2472348/b5f0ec7b-0a71-439c-b57f-28a1b620e771


---
